python/src/Flight.cpp

```diff
@@ -53,11 +53,24 @@ PyObject* xcsoar_Flight_new(PyTypeObject *type, PyObject *args, PyObject *kwargs
   self = (Pyxcsoar_Flight *)type->tp_alloc(type, 0);
   self->filename = nullptr;
 
+#if PY_MAJOR_VERSION >= 3
+  if (PyUnicode_Check(py_input_data)) {
```

Comment:
Wouldn't it be nice to allow byte strings in Python 3, too?

Reply:
It would be nice, yes. My main goal was to make the py-xcsoar test suite work though and it seems to be happy enough with the changes in the PR.